internal/search/query/parser.go

```
@@ -119,8 +119,22 @@ const (
 )
 
 func isSpace(buf []byte) bool {
+	for i := 0; i < len(buf); i++ {
```

Could you explain this part? What did we change in terms of handling spaces?

---

All except one call site call this with a single byte. For those call sites this is a NOP change.

In `matchKeyword` we use this to check whether "AND" or "OR" are the first tokens in the query. Before, we just checked whether "AND" or "OR" are at position 0. This means adding space in front of a keyword turned it from a pattern into a keyword, which is odd. 

The change I made just makes the treatment of keywords as first token consistent. However, I think we should rather return a error for this in the future.